networkdb/delegate.go

```diff
@@ -66,7 +66,9 @@ func (nDB *NetworkDB) handleNetworkEvent(nEvent *NetworkEvent) bool {
 		ltime: nEvent.LTime,
 	}
 
+	nDB.Lock()
```

Comment:
This lock is not needed because there is a lock/defer in the beginning of the function. This seems to be the cause of the deadlock